labs/SAW/SAW.md

```diff
@@ -41,30 +41,106 @@ tools. As well, there are some examples
 
+# Why Use SAW?
+
+Previous labs demonstrated how to look at algorithm specifications
+from math papers and formally represent them in Cryptol.  A
+software/hardware developer now can create a performant implementation
+with confidence since they are armed with a bit-precise, unambiguous,
+machine-readable, formally verified Cryptol specification.
+
+Of course, any developer desires that the code they write completely
+adheres to all of the cryptographic properties pertaining to the
+target algorithm. They also want to uphold best coding practices for
+their implementation language and avoid vulnerabilities such as buffer
+or integer overflows. After all, what good is cryptography if an
+attacker can take advantage of an implementation flaw?
+
+This is where the Software Analysis Workbench (SAW) comes in. SAW is a
+tool that formally verifies properties of compiled code using
+[automated reasoning
+tools](https://www.amazon.science/blog/a-gentle-introduction-to-automated-reasoning). Alright,
+that's a lot of fancy words, but essentially SAW can, in a fairly
+push-button way, verify that, for all inputs, a function does what
+it's supposed to do and nothing more. This makes SAW a much more
+powerful tool for assuring code compared to unit tests, which are
+limited in their testing scope and cannot test for all possible,
+potentially dangerous edge cases.
+
+In order to get SAW to work its magic, source code needs to be
+compiled - SAW works on low-level implementation languages such as
+[LLVM bitcode](https://en.wikipedia.org/wiki/LLVM), [Java byte
+code](https://en.wikipedia.org/wiki/Java_bytecode), low-level Verilog,
+and even some assembly languages. The examples in this lab demonstrate
+verification of cryptographic implementations written in the C
+language, so below you'll see C source code being compiled to LLVM
+bitcode.
+
+Next, we need to tell SAW exactly what to verify. For every
+function/method, SAW needs to know:
+- Input(s)
+- Output(s)
+- Precondition(s)
+- Postcondition(s)
+
+This information is packaged into a script for SAW to process. This
+script can either be written in Python or SAW's special purpose
+language, SAWScript. For this lab, the scripts are written in Python
+since, as a general purpose programming language, it provides a wider
+range of techniques when expressing properties and it is much more
+widely known than SAWScript. Python also fairly easily integrates into
+CI systems.
+
+Combining all of these different languages (Python, C, LLVM,
+SAWScript, Cryptol) and tools together into one workflow is definitely
+a challenging process, especially for newcomers. However, learning
+how to use these components provides a very powerful method for
+developing cryptographic implementations and proving them correct and
+secure.
+
+Before setting up the lab, let's recap the overall SAW workflow using the
+diagram below.
+
+![SAW Workflow Overview](../../misc/SAWWorkflowOverview.png)

```

Comment:
Hmm...  Is the Software Assurance Workbench more like a hand saw or a circular saw...?

Reply:
A hand saw just seemed fun to design as a placeholder at the time, but I suppose it depends on how much effort you have to put into using the tool :)

Sometimes the setup is quick and feels automatic like a circular saw. Sometimes the setup takes a lot more work by the user like a hand saw.

Regardless of the setup, SAW's magic does seem to whirr like a circular saw, so I think a circular saw would be better for an official logo.